Unable to compile the library on VisualStudio2019

Hi all, I'm trying to build `manif` on `VisualStudio2019`. When I configure the cmake project I get the following error:
```
Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.17134.
The CXX compiler identification is MSVC 19.26.28806.0
Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.26.28801/bin/Hostx64/x64/cl.exe
Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.26.28801/bin/Hostx64/x64/cl.exe -- works
Detecting CXX compiler ABI info
Detecting CXX compiler ABI info - done
Detecting CXX compile features
Detecting CXX compile features - done
Performing Test COMPILER_SUPPORTS_CXX11
Performing Test COMPILER_SUPPORTS_CXX11 - Failed
CMake Error at CMakeLists.txt:15 (message):
  The compiler C:/Program Files (x86)/Microsoft Visual
  Studio/2019/Enterprise/VC/Tools/MSVC/14.26.28801/bin/Hostx64/x64/cl.exe has
  no C++11 support.  Please use a different C++ compiler.


Configuring incomplete, errors occurred!
See also "C:/Users/gromualdi/robot-code/manif/build/CMakeFiles/CMakeOutput.log".
See also "C:/Users/gromualdi/robot-code/manif/build/CMakeFiles/CMakeError.log".
```


Associated issue: https://github.com/GiulioRomualdi/lie-group-controllers/pull/1#issuecomment-667695626

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,13 +7,18 @@ string(TOUPPER ${PROJECT_NAME} PROJECT_NAME_CAPS)
 
 # Check that the compoiler has c++11 support
 # Crash otherwise.
-include(CheckCXXCompilerFlag)
-CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
-if(COMPILER_SUPPORTS_CXX11)
-  message(STATUS "The compiler ${CMAKE_CXX_COMPILER} has C++11 support.")
+
+if(MSVC)
+  set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
 else()
-  message(FATAL_ERROR "The compiler ${CMAKE_CXX_COMPILER} "
-    "has no C++11 support. Please use a different C++ compiler.")
+  include(CheckCXXCompilerFlag)
+  CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
+  if(COMPILER_SUPPORTS_CXX11)
+    message(STATUS "The compiler ${CMAKE_CXX_COMPILER} has C++11 support.")
+  else()
+    message(FATAL_ERROR "The compiler ${CMAKE_CXX_COMPILER} "
+      "has no C++11 support. Please use a different C++ compiler.")
+  endif()
 endif()
 
 # Finding Eigen is somewhat complicated.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,9 +8,7 @@ string(TOUPPER ${PROJECT_NAME} PROJECT_NAME_CAPS)
 # Check that the compoiler has c++11 support
 # Crash otherwise.
 
-if(MSVC)
-  set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
-else()
+if(NOT MSVC)
   include(CheckCXXCompilerFlag)
   CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
   if(COMPILER_SUPPORTS_CXX11)

